### ISSUE TITLE
Additional multiple selection (multi-cursor) shortcuts

### ISSUE THREAD

**Issue by ericvoid** [notepadqq/notepadqq#950]
I commonly use the multiple selection (multiple cursors) shortcuts of text editors and IDEs which miss on notepadqq.

* Add Line above/below: adds a new cursor on the line above/below at the same column of the first cursor, or at the end of the line if it has fewer columns (issue #758 is pretty similar to this, but adding a line is not necessarily meant to create a block selection);
* Split Selection Into Lines: when a selection spans multiple lines, this features places a cursor at the end of each selected line;
* Quick Add Next: search for the selected text, and places a new selection/cursor on it (one occurrence at a time). This is very useful to rename variables, function names, etc that are close;
* Quick Find All: search for the selected text on all file contents and add a selection/cursor on every match (the same as Ctrl+F then "Select All", but using a single hotkey and without opening the search dialog).

All these features are available on SublimeText (https://www.sublimetext.com/docs/3/multiple_selection_with_the_keyboard.html) and many of them are available on VSCode (https://code.visualstudio.com/docs/editor/codebasics#_multiple-selections-multicursor)

**Comment by mbomb007** [comment]
Yeah, I really would like to be able to middle-click+drag down a column, then remove all of that and to the left. Or to multi-copy/paste  pieces of lines. VS Code can do this, and it's really helpful.
